DirectProgramming/DPC++FPGA/Tutorials/Features/device_global/README.md

```diff
@@ -0,0 +1,225 @@
+# `device_global`
+This FPGA tutorial explains how to use `device_global` class as a way of keeping state between multiple kernel invocations.
+
+| Optimized for                     | Description
+|:---                                 |:---
+| OS                                | Linux* Ubuntu* 18.04/20.04 <br> RHEL*/CentOS* 8 <br> SUSE* 15 <br> Windows* 10
+| Hardware                          | Intel&reg; Programmable Acceleration Card (PAC) with Intel Arria&reg; 10 GX FPGA; <br> Intel&reg; FPGA Programmable Acceleration Card (PAC) D5005 (with Intel Stratix&reg; 10 SX) <br> Intel&reg; FPGA 3rd party / custom platforms with oneAPI support <br> **Note**: Intel&reg; FPGA PAC hardware is only compatible with Ubuntu 18.04*
+| Software                          | Intel&reg; oneAPI DPC++ Compiler <br> Intel&reg; FPGA Add-On for oneAPI Base Toolkit
+| What you will learn               | The basic usage of the `device_global` class <br> How initialize a `device_global` to non-zero values<br>
+| Time to complete                  | 15 minutes
+
+## Purpose
+This tutorial demonstrates a simple example of initializing a `device_global` to a non-zero value, and using it to keep state between multiple re-launches of a kernel.
+
+### Description of `device_global`
+`device_globals` are an extension that introduces device scoped memory allocations into SYCL that can be accessed within a kernel using syntax similar to C++ global variables, but that have unique instances per `sycl::device`. Similar to C++ global variables, `device_globals` have namespace scope and are visible to all kernels within that scope.
+
+`device_globals` can be templated on a list of compile-time-constant `properties` that can enable various optimization behaviors. One such `property` is `device_image_scope` which is used in the code sample. It limits the scope of a single instance of a `device_global` to a `device_image`, further details on this property can be found [here](https://github.com/intel/llvm/blob/sycl/sycl/doc/extensions/proposed/sycl_ext_oneapi_device_global.asciidoc#properties-for-device-global-variables). The absence of the `device_image_scope` property is currently not supported by the compiler.
+
+`device_globals` can be used to store state across multiple re-launches of a kernel without having to pass in a `buffer` as a kernel argument. This is useful in cases where components are nodes in a state-machine.
+
+### Non-zero-initialization of a `device_global` - pre C++20
+Without C++20 `consteval` keyword support, `device_globals` are zero-initialized. If there is a known first usage of a `device_global` in device code, there exists a technique, described below, that allows a `device_global` to be initialized to a non-zero value. 
```

Comment:
My intent was to rely on the C++ terminology of [zero-initialization](https://en.cppreference.com/w/cpp/language/zero_initialization) and [value-initialization](https://en.cppreference.com/w/cpp/language/value_initialization]) so that it is intuitive to the reader what is the behavior/benefits of each. 

What will help in order to clarify the situation. Again i want to avoid truing to define these terms here because they are complicated and have nuances depending on the type that is being initialized.

Reply:
Yeah, I would personally just says "note that the technique below for initializing the `device_global` variables will be simplified once we support C++20 to `device_global<int> a {42};`"